modin/pandas/test/test_dataframe.py

```diff
@@ -4460,6 +4461,25 @@ def test___setitem__(self, data):
         modin_df = pd.DataFrame(data)
         pandas_df = pandas.DataFrame(data)
 
+        modin_df[modin_df.columns[-1]] = pd.DataFrame(modin_df[modin_df.columns[0]])
+        pandas_df[pandas_df.columns[-1]] = pandas.DataFrame(
+            pandas_df[pandas_df.columns[0]]
+        )
+        df_equals(modin_df, pandas_df)
+
+        modin_df = pd.DataFrame(data)
+        pandas_df = pandas.DataFrame(data)
+
+        # TODO: These commented out lines are some nasty edge cases we don't support
```

Comment:
What is stopping us from supporting these edge cases?

Reply:
We could if statement this out, but it's unclear/undocumented when shapes should be coerced. I'm mostly not sure if there's an underlying reason why there should be different behavior between setting a nonexistent column vs. one that exists